curriculum/roadmap/engine.py

```
@@ -0,0 +1,75 @@
+                                hours_per_week, level, feedback)
+        payload, usage = llm.generate(text)
+        tok_in += usage.get("promptTokenCount", 0)
+        tok_out += usage.get("candidatesTokenCount", 0)
+
+        selected = payload.get("selected", [])
+        valid, total, problems = verify.check(courses, selected, budget)
+
+        if not problems or attempt == max_attempts:
+            reasons = [s.get("reason", "") for s in selected
+                       if 0 <= s["index"] < len(courses)]
+            pairs = list(zip(valid, reasons))
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)